crates/matrix-sdk/src/room/timeline/mod.rs

```diff
@@ -389,6 +391,36 @@ impl Timeline {
         };
         self.inner.update_event_send_state(&txn_id, send_state);
     }
+
+    /// Fetch unavailable details about the event with the given ID.
+    ///
+    /// This method only works for IDs of [`RemoteEventTimelineItem`]s, to
+    /// prevent losing details when a local echo is replaced by its remote
+    /// echo.
+    ///
+    /// This method tries to make all the requests it can. If an error is
+    /// encountered for a given request, it is forwarded with the
+    /// [`TimelineDetails::Error`] variant.
+    ///
+    /// # Arguments
+    ///
+    /// * `event_id` - The event ID of the event to fetch details for.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the identifier doesn't match any event with a remote
+    /// echo in the timeline, or if the event is removed from the timeline
+    /// before all requests are handled.
+    #[instrument(fields(room_id = ?self.room().room_id()))]
```

Comment:
Logging `self` is probably going to be very noisy.

```suggestion
    #[instrument(skip(self), fields(room_id = ?self.room().room_id()))]
```